src/commonTest/kotlin/fr/acinq/lightning/serialization/StateSerializationNonRegTestsCommon.kt

```diff
@@ -36,6 +38,20 @@ class StateSerializationNonRegTestsCommon {
                 if (debug) {
                     tmpFile.delete()
                 }
+                val state1 = when (state) {
+                    is LegacyWaitForFundingConfirmed -> state.copy(commitments = state.commitments.copy(remoteChannelData = EncryptedChannelData.empty))
+                    is LegacyWaitForFundingLocked -> state.copy(commitments = state.commitments.copy(remoteChannelData = EncryptedChannelData.empty))
+                    is WaitForFundingConfirmed -> state.copy(commitments = state.commitments.copy(remoteChannelData = EncryptedChannelData.empty))
+                    is WaitForChannelReady -> state.copy(commitments = state.commitments.copy(remoteChannelData = EncryptedChannelData.empty))
+                    is Normal -> state.copy(commitments = state.commitments.copy(remoteChannelData = EncryptedChannelData.empty))
+                    is ShuttingDown -> state.copy(commitments = state.commitments.copy(remoteChannelData = EncryptedChannelData.empty))
+                    is Negotiating -> state.copy(commitments = state.commitments.copy(remoteChannelData = EncryptedChannelData.empty))
+                    is Closing -> state.copy(commitments = state.commitments.copy(remoteChannelData = EncryptedChannelData.empty))
+                    is WaitForRemotePublishFutureCommitment -> state.copy(commitments = state.commitments.copy(remoteChannelData = EncryptedChannelData.empty))
+                    is Closed -> state
+                }
+                val state2 = Serialization.deserialize(Serialization.serialize(state))
+                assertEquals(state1, state2, path.toString())
```

Comment:
More uglyness. We can't just compare `state` with `state2` because test data was generated using `lightning-kmp` integration tests, which have two Phoenix peers, which cause them to both have a `remoteChannelData` (which will be dropped at serialization).

I could regenerate the test data, or simpler (and maybe better?): serialize `remoteChannelData`. Why not after all?